core/node/rpc/sync/client/remote.go

```diff
@@ -0,0 +1,257 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"connectrpc.com/connect"
+	"github.com/ethereum/go-ethereum/common"
+	. "github.com/river-build/river/core/node/base"
+	"github.com/river-build/river/core/node/dlog"
+	. "github.com/river-build/river/core/node/protocol"
+	"github.com/river-build/river/core/node/protocol/protocolconnect"
+	. "github.com/river-build/river/core/node/shared"
+)
+
+type remoteSyncer struct {
+	syncStreamCtx    context.Context
+	syncStreamCancel context.CancelFunc
+	syncID           atomic.Value
```

Comment:
it seems it doesn't have to be atomic: it is written once during object creation?